METHOD_NAMING_PROPOSAL.md

```diff
@@ -0,0 +1,412 @@
+}
+```
+
+</td>
+
+</tr>
+<tr>
+<th>v4 3DS</th>
+<th>v5 3DS</th>
+</tr>
+<tr>
+
+<td>
+
+```kotlin 
+// Retrieve data required to start 3DS authentication
+// Or return result if no additional auth required
+threeDSecureClient.performVerification(activity, request) { threeDSecureResult, error ->
+  // Launch 3DS flow in Cardinal SDK 
+  // Return nonce to listener
+  threeDSecureClient.continuePerformVerification(activity, request, threeDSecureResult)
+}
+```
+
+</td>
+
+<td>
+
+```kotlin
+threeDSecureLauncher = ThreeDSecureLauncher() { authenticationResult ->
+  // Parse the result from the 3DS auth flow and return a nonce
+  threeDSecureClient.continuePerformVerification(authenticationResult) 
+}
+
+// Retrieve data required to launch the 3DS flow
+// Instantiate Cardinal SDK
+threeDSecureClient.performVerification(context, request) { threeDSecureResult, error ->
+  // Launch 3DS flow in Cardinal SDK 
+  threeDSecureLauncher.launch(threeDSecureResult) 
+}
+```
+
+</td>
+
+</tr>
+
+<tr>
+<th>v4 Venmo</th>
+<th>v5 Venmo</th>
+</tr>
+<tr>
+
+<td>
+
+```kotlin 
+// Create Venmo payment context
+// Launch Venmo app and create nonce from payment context app switch result
+// Return nonce to listener
+venmoClient.tokenizeVenmoAccount(activity, request)
+```
+
+</td>
+
+<td>
+
+```kotlin
+venmoLauncher = VenmoLauncher(activity) { authChallengeResult ->
+  // Create nonce from payment context app switch result and vault if required
+  venmoClient.tokenizeVenmoAccount(authChallengeResult) 
+}
+
+
+  // Create Venmo payment context
+venmoClient.requestAuthChallenge(context, request) { authChallenge, error ->
+  // Launch Venmo app
+  venmoLauncher.launch(authChallenge) 
+}
+```
+
+</td>
+
+</tr>
+
+<tr>
+<th>v4 Google Pay</th>
+<th>v5 Google Pay</th>
+</tr>
+<tr>
+
+<td>
+
+```kotlin 
+// Create PaymentDataRequest for Google Pay SDK
+// Launch Google Pay payment sheet
+// Parse Google Pay result an return a nonce
+googlePayClient.requestPayment(activity, request)
+```
+
+</td>
+
+<td>
+
+```kotlin
+googlePayLauncher = GooglePayLauncher(activity) { googlePayResult ->
+  // Parse Google Pay result and return a nonce
+  googlePayClient.tokenize(googlePayResult) 
+}
+
+// Create PaymentDataRequest for Google Pay SDK
+googlePayClient.requestPayment(context, request) { googlePayIntentData, error ->
+  // Launch Google Pay payment sheet
+  googlePayLauncher.launch(googlePayIntentData) 
+}
+```
+
+</td>
+
+</tr>
+</table>
+
+
+### Consistent Naming Pattern
+
+With this implementation, the method names do not align with v4, so there is greater merchant migration difficulty, but the naming pattern is consistent across payment methods and provides clarity to what is actually occurring.
+For this pattern, I've chosen to use the `paymentAuth` naming pattern.
+
+`client.createPaymentAuthRequest` returns `paymentAuthRequest`
+
+`launcher.launch(paymentAuthRequest)` returns `paymentAuthResult`
+
+`client.tokenize(paymentAuthResult)` returns `paymentMethodNonce`
```

Comment:
This may be a 🔥 take but do we need to use the work `auth` at all? Could we do:
```suggestion
`client.createPaymentRequest` returns `paymentRequest`

`launcher.launch(paymentRequest)` returns `paymentResult`

`client.tokenize(paymentResult)` returns `paymentMethodNonce`
```
Mainly thinking from the point of view that `auth` tends to signal an authorization attempt vs capturing a payment which could be confusing? Not sure what you/others think, so just throwing some ideas out there.

Reply:
I think of Auth as permission, so it makes sense to me but I don't feel too strongly about it. Less change for typo is good also.

Reply:
It would be helpful for me to see naming change proposals side by side (what is now, minimum change, consistent changes) but I think you all probably have them memorized.

Reply:
@jaxdesmarais I like plain `paymentRequest` too! 

@KunJeongPark You should be able to see the side by side comparison if you view the rich diff of the markdown file (I tried to make a table with two columns). The v4 column is what is now, and the v5 column is the proposed change)

Reply:
😈 advocate - `PaymentResult` might be easy to confuse with `PaymentMethodNonce`

Reply:
I like the names you have proposed, Sarah! I can't think of anything better ATM

Reply:
Another option would be to name the request/result pair to highlight the integration point of `Launcher` and `Client`.  `LauncherRequest` and `LauncherResult` might be helpful in that case. I do prefer the naming convention to be decoupled from the implementation though, that would be ideal.

Overall, the best name will make it clear that additional steps are required by the buyer, and the result of the auth challenge will be handled by the SDK before a `PaymentMethodNonce` is returned.